MonoGame.Framework.Content.Pipeline/Builder/PipelineManager.cs

```diff
@@ -716,65 +797,120 @@ public object ProcessContent(PipelineBuildEvent pipelineEvent)
                     }
-
-                    CleanContent(string.Empty, asset);
                 }
 
-                // Remove related output files (non-XNB files) that were copied to the output folder.
-                foreach (var asset in cachedEvent.BuildOutput)
+                if (cachedEvent != null)
                 {
-                    Logger.LogMessage("Cleaning {0}", asset);
-                    FileHelper.DeleteIfExists(asset);
-                }
-            }
+                    // Recursively clean additional (nested) assets.
+                    foreach (var asset in cachedEvent.BuildAsset)
+                    {
+                        string assetEventFilepath;
+                        var assetCachedEvent = LoadBuildEvent(asset, out assetEventFilepath);
 
-            Logger.LogMessage("Cleaning {0}", outputFilepath);
+                        if (assetCachedEvent == null)
+                        {
+                            Logger.LogMessage("Cleaning {0}", asset);
 
-            // Remove asset (.xnb file) from output folder.
-            FileHelper.DeleteIfExists(outputFilepath);
+                            // Remove asset (.xnb file) from output folder.
+                            FileHelper.DeleteIfExists(asset);
 
-            // Remove event file (.mgcontent file) from intermediate folder.
-            FileHelper.DeleteIfExists(eventFilepath);
+                            // Remove event file (.mgcontent file) from intermediate folder.
+                            FileHelper.DeleteIfExists(assetEventFilepath);
+                            continue;
+                        }
+
+                        CleanContent(asset);
+                    }
 
-            _pipelineBuildEvents.Remove(sourceFilepath);
+                    // Remove related output files (non-XNB files) that were copied to the output folder.
+                    foreach (var asset in cachedEvent.BuildOutput)
+                    {
+                        Logger.LogMessage("Cleaning {0}", asset);
+                        FileHelper.DeleteIfExists(asset);
+                    }
+
+                    // jcf: clean intermediate output???
+                }
+            }
+            finally
+            {
+                Logger.Unindent();
+                Logger.PopFile();
+            }
+
+            return clean;
         }
 
         private void WriteXnb(object content, PipelineBuildEvent pipelineEvent)
         {
             // Make sure the output directory exists.
             var outputFileDir = Path.GetDirectoryName(pipelineEvent.DestFile);
+            
+            var wrongCase = PathHelper.DirectoryExistsWithDifferentCase(outputFileDir);
+            if (wrongCase.HasValue && wrongCase.Value == true)
+            {
+                throw new Exception(string.Format("Failed to write xnb '{0}' because the path already exists in a different case!", pipelineEvent.DestFile));
```

Comment:
Yes, that is the intention. Think of asset name as what you would pass into Content.Load(...) at runtime. It is the filename and path relative to the content root.

In xna what you saw in the IDE was just the 'name'. But it did know the input and output paths relative to the project root / output root (respecively). Because this is how including an item "as a link" worked.